### PR TITLE
[Yext] refactoring and Oauth flow

### PR DESCRIPTION
**Pull request by EugeneLightsOn**
### What's being changed:

- Refactors client code into client.py
- Oauth flow
- Clarifies code and improve README

### How did you test this change (include any code snippets, API requests, screenshots, or gifs):
image

### CODE REVIEW

File: yext/README.md [hunk 1]
```diff
@@ -4,28 +4,108 @@ Connects Cohere to Yext Universal Search.
...
+
+```
+YEXT_SEARCH_LIMIT
+```
+This variable may contain a JSON object specifying the limit for each vertical. 
+Each key is a vertical key and the value for each of those keys is a number 1-50 that denotes the limit for that vertical. 
+If a vertical is not specified, the default limit is 10 for all verticals
+
+```
+YEXT_RESTRICT_VERTICALS
```
Comment (tianjing-li) [inline yext/README.md:91]: Both the `SEARCH_LIMIT` and `RESTRICT_VERTICALS` variables mention verticals, I would either document the full list of verticals available by Yext here for clarity or link the documentation, to me it seems unclear what possible options I can set

**Review by tianjing-li — APPROVED** [review]
Minor changes, looks great

File: yext/README.md [hunk 2]
```diff
@@ -4,28 +4,108 @@ Connects Cohere to Yext Universal Search.
...
+
+```
+YEXT_LOCALE
+```
+This variable should be set to the locale of the search experience you want to use. This is typically `en`.
+
+```
+YEXT_V
+```
+This variable should be set to the search version date in YYYYMMDD format. You can find this in the Yext UI under Search Experience >Test search.
```
[the remark this answers is not in the document]
Comment (EugeneLightsOn) [reply yext/README.md]: Tank you

File: yext/README.md (hunk 1 above)
Comment (EugeneLightsOn) [reply yext/README.md:92]: Thanks

**Pull request merged**